Mathlib/NumberTheory/SiegelsLemma.lean

```diff
@@ -0,0 +1,237 @@
+/-
+Copyright (c) 2024 Fabrizio Barroero. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fabrizio Barroero, Laura Capuano, Amos Turchet
+-/
+import Mathlib.Analysis.Matrix
+import Mathlib.Data.Pi.Interval
+import Mathlib.Tactic.Rify
+
+/-!
+# Siegel's Lemma
+
+In this file we introduce and prove Siegel's Lemma in its most basic version. This is a fundamental
+tool in diophantine approximation and transcendency and says that there exists a "small" integral
+non-zero solution of a non-trivial underdetermined system of linear equations with integer
+coefficients.
+
+## Main results
+
+- `exists_ne_zero_int_vec_norm_le`: Given a non-zero `m × n` matrix `A` with `m < n` the linear
+system it determines has a non-zero integer solution `t` with
+`‖t‖ ≤ ((n * ‖A‖) ^ ((m : ℝ) / (n - m)))`
+
+## Notation
+
+ - `‖_‖ ` : Matrix.seminormedAddCommGroup is the sup norm, the maximum of the absolute values of
+ the entries of the matrix
+
+## References
+
+See [M. Hindry and J. Silverman, Diophantine Geometry: an Introduction][hindrysilverman00].
+-/
+
+/- We set ‖⬝‖ to be Matrix.seminormedAddCommGroup  -/
+attribute [local instance] Matrix.seminormedAddCommGroup
+
+open Matrix Finset
+
+section generalMatrices
+
+variable {m n α : Type*} [Fintype m] [Fintype n] [SeminormedAddCommGroup α]
+
+lemma exists_entry_norm_eq_norm_Matrix [Nonempty m] [Nonempty n] (A : Matrix m n α) :
+    ∃ (i : m) (j : n), ‖A‖ = ‖A i j‖₊ := by
+  simp only [norm_eq_sup_sup_nnnorm]
+  rcases Finset.exists_mem_eq_sup univ univ_nonempty (fun i => univ.sup fun j => ‖A i j‖₊)
+    with ⟨i, hi⟩
+  use i
+  rcases Finset.exists_mem_eq_sup univ univ_nonempty (fun j => ‖A i j‖₊) with ⟨j, hj⟩
+  use j
+  rw [hi.2, hj.2]
```

Comment:
```suggestion
  simp only [norm_eq_sup_sup_nnnorm]
  obtain ⟨i, hi⟩ := Finset.exists_mem_eq_sup _ univ_nonempty (univ.sup fun j => ‖A · j‖₊)
  obtain ⟨j, hj⟩ := Finset.exists_mem_eq_sup _ univ_nonempty (‖A i ·‖₊)
  exact ⟨i, j, by rw [hi.2, hj.2]⟩
```
and yes, please move this lemma out, to where you suggested to. Than we can revise the whole Siegel part. Thanks!

Reply:
I think Mathlib style is to use `↦` (\mapsto) instead of `=>` with `fun`?

Reply:
I see this comment has unfortunately not been taken into account. It is not a great deal, but apart from the golfing I believe this lemma should not stay there but be mobed to `Analysis.Matrix`.